### ISSUE TITLE
Return list of products (even if empty)

### ISSUE THREAD

**Issue by rafaelcmrj** [PeterStaev/nativescript-purchase#26]
There is no way to know if the user has purchased or not the product. The restorePurchase method only fire the event IF the user has made a purchase. How to validate if the user has bought or not a product?

**Comment by PeterStaev** [comment]
Hey @rafaelcmrj , there is no way to get this. You should work with the `restorePurchases` available functions. For example your app can initialize an empty array and save it in local settings. Then you can call `restorePurcahses` and in the callback add the bought products to the array. Then in the rest of your application, where you want to check if a product is bought, you can check the array in the local settings to see if the product is there or not. Hope this helps.

**Comment by rafaelcmrj** [comment]
@PeterStaev the problem is that the `restorePurchases` method doesn't call the callback if you have no products in your account, so I will never know if the method was executed or not... The correct way to do that was to put my application initializer inside the callback function, so I will make sure I only start my application AFTER verify if you have or not the purchase. Makes sense?

**Comment by PeterStaev** [comment]
@rafaelcmrj this is not how I would implement it. Like I have outlined in the step by step guide in the readme, you should pertain info about what products are purchased and what not in the `application-settings` provided by NativeScript. Once your user purchases a product you should save this locally in the application settings, so you can use it directly after the purchase is complete. You should not call `resotrePurcahses` every time you want to know if the user bought a product or not. `restorePurchases` should be triggered by the user and is basically usable only when the user uninstall the app or installs it on another phone. In this case they trigger purchase restoring and in the callback you again set the application settings just like a normal purchase. 

So to sum it up - your app should start in "unpaid" mode with all paid functions hidden/disabled. If the user buys a product you save this in the `application-settings` and enable the corresponding function. If the user changes their phone they press some button/menu to restore purchases. Then you wait in the callback and if you receive info about some product you save that again in the `application-settings` and again enable the paid function for that product.

**Comment by PeterStaev** [comment]
@rafaelcmrj this is not how I would implement it. Like I have outlined in the step by step guide in the readme, you should pertain info about what products are purchased and what not in the `application-settings` provided by NativeScript. Once your user purchases a product you should save this locally in the application settings, so you can use it directly after the purchase is complete. You should not call `resotrePurcahses` every time you want to know if the user bought a product or not. `restorePurchases` should be triggered by the user and is basically usable only when the user uninstall the app or installs it on another phone. In this case they trigger purchase restoring and in the callback you again set the application settings just like a normal purchase. 

So to sum it up - your app should start in "unpaid" mode with all paid functions hidden/disabled. If the user buys a product you save this in the `application-settings` and enable the corresponding function. If the user changes their phone they press some button/menu to restore purchases. Then you wait in the callback and if you receive info about some product you save that again in the `application-settings` and again enable the paid function for that product.

**Comment by rafaelcmrj** [comment]
@PeterStaev I was doing on this way, but if we only check the app-settings, how can we guarantee that the user didn't cancel the subscription after he/she activated the product? Doing in this way, someone can buy a product, activate the new version and cancel the purchase. He will be able to access the paid version after canceling the product.

**Comment by PeterStaev** [comment]
Subscription are not yet supported by the plugin (#2). And from what I know a normal in-app product cannot be cancelled.

**Comment by erjdriver** [comment]
I've found that non-renewable subscriptions do work - most of the time.

Sometimes - I don't get some purchase notifications - but that could be something on this end.

But - yes users can purchase and then cancel and there's no way to know of that. Apple iTunes doesn't give you any info on such transactions.
